app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -68,17 +68,15 @@ def change_datacenter_to_uuid(datacenter)
 
       def create
         begin
-          if compute_resource_params["provider"].downcase == "ovirt"
+          if compute_resource_params["provider"].downcase == "ovirt" && !compute_resource_params[:datacenter].is_uuid?
             @compute_resource = ComputeResource.new_provider(compute_resource_params.except(:datacenter))
-          else
-            @compute_resource = ComputeResource.new_provider(compute_resource_params)
+            compute_resource_params[:datacenter] = change_datacenter_to_uuid(compute_resource_params[:datacenter])
```

Comment:
a, i think i know why this is failing - `compute_resource_params` is a wrapper for the permitted request params, we should be checking and modifying `params[:datacenter]` and `params[:provider]` instead, compute_resource_params should only be used for mass assignment (such as in the new_provider method)

Reply:
in other words, something like this:
```ruby
          if params[:provider].downcase == "ovirt" && !params[:datacenter].is_uuid?
              dc = params.delete(:datacenter)
              @compute_resource = ComputeResource.new_provider(compute_resource_params)
              params[:datacenter] = change_datacenter_to_uuid(dc)
          end

Reply:
thanks @tbrisker but it's not the case, it was failing because of: 
!params[:datacenter].is_uuid? 
instead, it should be something like: 
!Foreman.is_uuid?(compute_resource_params[:datacenter])

now, I have another failure that I need to figure out.

Reply:
sorry, @tbrisker  you were right, adding the changes.